storage/factory.go

```diff
@@ -18,25 +18,30 @@ import (
 	"github.com/jaegertracing/jaeger/storage/spanstore"
 )
 
+// Same as Factory, but without the Initialize method.
+// It was a design mistake originally to add Initialize to the Factory interface.
+type BaseFactory interface {
```

Comment:
do we have a longer term plan to get rid of `Factory` and replace it with this?

Reply:
I don't know if it's worth it for v1. We should make sure v2 API is not repeating the same mistake.